src/lv_draw/lv_draw_img.c

```diff
@@ -424,6 +420,10 @@ bool lv_img_color_format_has_alpha(lv_img_cf_t cf)
     switch(cf) {
         case LV_IMG_CF_TRUE_COLOR_ALPHA:
         case LV_IMG_CF_RAW_ALPHA:
+        case LV_IMG_CF_INDEXED_1BIT:
+        case LV_IMG_CF_INDEXED_2BIT:
+        case LV_IMG_CF_INDEXED_4BIT:
+        case LV_IMG_CF_INDEXED_8BIT:
```

Comment:
`LV_INDEXED_CHROMA` guard here too?

Reply:
It's not needed - there's no harm in leaving transparency on if chroma is in use.

Reply:
See the [comment above](https://github.com/littlevgl/lvgl/pull/1183#discussion_r318602446)

Reply:
I have left this as is. The behavior of `LV_INDEXED_CHROMA` in the pull-request is as follows:

- 0: Indexed images have transparency
- 1: Indexed images have transparency and a chroma color.

With your proposed change, `LV_INDEXED_CHROMA` will operate as follows:
- 0: Indexed images have transparency
- 1: Indexed images have a chroma color (and no transparency).

I have made `LV_INDEXED_CHROMA` 1 by default to mimic the original behavior. By implementing this change, transparency for indexed images will be off by default, which I think is removing a useful feature from the default configuration.

I can make this change if you wish as I'm always going to use `LV_INDEXED_CHROMA 0`, but I think it's best to leave as is.

Reply:
As I see if `LV_INDEXED_CHROMA`
- 0: Indexed images have "real" transparency with an alpha byte
- 1: Alpha byte is not used, but `LV_COLOR_TRANSP` pixels will be transparent (choma keyed)

It's two different way to handle transparency and they can't be mixed. (both shouldn't be used at the same time).

> I have made LV_INDEXED_CHROMA 1 by default to mimic the original behavior.

That's great!

> I can make this change if you wish as I'm always going to use LV_INDEXED_CHROMA 0, but I think it's best to leave as is.

Let's discuss it first. Maybe I missed somethings so it'd be better if we agreed on the solution.

Reply:
> It's two different way to handle transparency and they can't be mixed. (both shouldn't be used at the same time).

I don't see the issue with mixing chroma keying and real transparency. The current image drawing code can handle them both set to true at the same time:

https://github.com/littlevgl/lvgl/blob/74d5ac55532dd2d3bd85ab5ebe087a9300f8a1a5/src/lv_draw/lv_draw_basic.c#L483-L504

As well as the canvas drawing code:

https://github.com/littlevgl/lvgl/blob/74d5ac55532dd2d3bd85ab5ebe087a9300f8a1a5/src/lv_objx/lv_canvas.c#L354-L419

Which are the only two places that `lv_img_color_format_is_chroma_keyed` and `lv_img_color_format_has_alpha` are used.

I understand your concern about mixing the two different transparency modes, so I'll explain why I've set it up that way. I wanted the default configuration to support "real" transparency in images so that it's easy to use, i.e. a new user can just upload an image with transparency to the converter, add it to the project and it just works. I also didn't want to break anyone's existing code where they were using chroma keyed images, so as a compromise I made the default support both "real" transparency and chroma keying.

The disadvantage of configuring it in the way you are suggesting is with the default configuration. With the default set to chroma, a new user may convert an image and find it appears opaque. With the default set to "real" transparency, existing users will find their chroma keyed images no longer working. The way I've configured it will solve both problems, and the user to turn off chroma keying if they want to use green in their "real" transparency image.

From my perspective, it would make most sense to just remove support for chroma keyed images for indexed images and use "real" transparency, but I'm not sure you'll find that suggestion agreeable!

Reply:
> I don't see the issue with mixing chroma keying and real transparency. The current image drawing code can handle them both set to true at the same time

Hmm, they are really working together. You are right and your suggestion really gives good flexibility and backward compatibility too. Let's do this way!

> From my perspective, it would make most sense to just remove chroma transparency support for indexed images and use "real" transparency, but I'm not sure you'll find that suggestion agreeable!

I agree it would be really better to use only "real" opacity. I'll update it in the next major release.